src/System.Globalization/tests/CultureInfo/CultureInfoAsync.cs

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Globalization.Tests
+{
+    public class CultureInfoAsync
+    {        
+        [Fact]
+        public void TestCurrentCulturesAsync()
+        {
+            CultureInfo currentCulture = CultureInfo.CurrentCulture;
+            CultureInfo currentUICulture = CultureInfo.CurrentUICulture;
+
+            CultureInfo newCurrentCulture = new CultureInfo(currentCulture.Name.Equals("ja-JP", StringComparison.OrdinalIgnoreCase) ? "en-US" : "ja-JP");
+            CultureInfo newCurrentUICulture = new CultureInfo(currentUICulture.Name.Equals("ja-JP", StringComparison.OrdinalIgnoreCase) ? "en-US" : "ja-JP");
+            
+            CultureInfo.CurrentCulture = newCurrentCulture;
+            CultureInfo.CurrentUICulture = newCurrentUICulture;
+            
+            Task.Run(() => {
+                Assert.Equal(CultureInfo.CurrentCulture, newCurrentCulture);
+                Assert.Equal(CultureInfo.CurrentUICulture, newCurrentUICulture);
+            }).Wait();
+
+            CultureInfo.CurrentCulture = currentCulture;
+            CultureInfo.CurrentUICulture = currentUICulture;
```

Comment:
These should be in a finally block, for a try block containing the Task.Run(...).Wait().  Otherwise if there's an assert failure, the culture won't get reset appropriately.